lib/device-settings.js

```diff
@@ -0,0 +1,38 @@
+import _ from 'lodash';
+import log from './logger';
+
+class DeviceSettings {
+
+  constructor(settings = {}) {
```

Comment:
also, let's call the local variable `defaultSettings` so it's more clear what this is